jupyter_archive/handlers.py

```diff
@@ -29,6 +30,13 @@ def __init__(self, handler):
         self.position = 0
 
     def write(self, data):
+        if self.handler.canceled:
+            raise ValueError("File download canceled")
+        while len(self.handler._write_buffer) > self.handler.handler_max_buffer_length:
```

Comment:
Could you add a time out condition that raises an error if hit? So we are sure this loop ends even if the buffer is not clean.